src/translate/pddl_parser/parsing_functions.py

```diff
@@ -611,15 +610,18 @@ def parse_init(context, alist):
                 if not isinstance(fact, list) or not fact:
                     context.error("Invalid negated fact.", syntax=SYNTAX_LITERAL_NEGATED)
                 atom = pddl.Atom(fact[0], fact[1:])
-                check_atom_consistency(context, atom, initial_false, initial_true, False)
-                initial_false.add(atom)
+                check_atom_consistency(context, atom,
+                                       initial_proposition_values, False)
+                initial_proposition_values[atom] = False
             else:
                 if len(fact) < 1:
                     context.error(f"Expecting {SYNTAX_LITERAL} for atoms.")
                 atom = pddl.Atom(fact[0], fact[1:])
-                check_atom_consistency(context, atom, initial_true, initial_false)
-                initial_true.add(atom)
-    initial.extend(initial_true)
+                check_atom_consistency(context, atom,
+                                       initial_proposition_values, True)
+                initial_proposition_values[atom] = True
+    initial.extend(atom for atom, val in initial_proposition_values.items()
+                   if val is True)
```

Comment:
It seems we set `initial` to the empty list, never modify it, then extend it here once. So there seems no reason to introduce `initial` before, and we can instead assign here rather than extend. But it's unrelated to the issue, so feel free to do it or leave it.

Reply:
In line 605 we insert the assignments to the numeric fluents to initial, so the extend  makes sense here.